protocol/xmrtaker/swap_state.go

```diff
@@ -390,16 +390,21 @@ func (s *swapState) exit() error {
 		txHash, err := s.tryRefund()
 		if err != nil {
 			if strings.Contains(err.Error(), revertSwapCompleted) {
-				return s.tryClaim()
+				// note: this should NOT ever error; it could if the ethclient
+				// or monero clients crash during the course of the claim,
+				// but that would be very bad.
+				err = s.tryClaim()
+				if err != nil {
+					panic(fmt.Sprintf("failed to claim even though swap was completed on-chain: %s", err))
```

Comment:
panic may be too extreme here. It looks like this could happen if, for example, the endpoints are overloaded ... or a user went past their API limit on an endpoint. It wouldn't take a full crash of the client.